runtime/jcl/common/mgmtgc.c

```diff
@@ -171,64 +178,59 @@ Java_com_ibm_lang_management_internal_ExtendedGarbageCollectorMXBeanImpl_getLast
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, preCommittedArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->preCommitted[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, preMaxArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->preMax[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, postUsedArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->postUsed[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, postCommittedArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->postCommitted[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-	(*env)->SetLongArrayRegion(env, postMaxArray, 0, gcInfo->arraySize, (jlong *) &gcInfo->postMax[0]);
-	if ((*env)->ExceptionCheck(env)) {
-		goto fail2;
-	}
-
-	for (idx = 0; idx < gcInfo->arraySize; ++idx) {
-		jstring poolName = NULL;
-		if (mgmt->supportedMemoryPools > idx) {
-			poolName = (*env)->NewStringUTF(env, mgmt->memoryPools[idx].name);
-		} else {
-			poolName = (*env)->NewStringUTF(env, mgmt->nonHeapMemoryPools[idx - mgmt->supportedMemoryPools].name);
+	{
+		jboolean isCopy = JNI_FALSE;
+		initialArrayElems = (jlong*) (*env)->GetPrimitiveArrayCritical(env, initialArray, &isCopy);
+		if (NULL == initialArrayElems) {
+			goto fail2;
 		}
-		if (NULL == poolName) {
+		preUsedArrayElems = (jlong*) (*env)->GetPrimitiveArrayCritical(env, preUsedArray, &isCopy);
+		if (NULL == preUsedArrayElems) {
 			goto fail2;
 		}
-
-		(*env)->SetObjectArrayElement(env, poolNamesArray, idx, poolName);
-		if ((*env)->ExceptionCheck(env)) {
+		preCommittedArrayElems = (jlong*) (*env)->GetPrimitiveArrayCritical(env, preCommittedArray, &isCopy);
+		if (NULL == preCommittedArrayElems) {
+			goto fail2;
+		}
+		preMaxArrayElems = (jlong*) (*env)->GetPrimitiveArrayCritical(env, preMaxArray, &isCopy);
+		if (NULL == preMaxArrayElems) {
 			goto fail2;
 		}
-		(*env)->DeleteLocalRef(env, poolName);
+		postUsedArrayElems = (jlong*) (*env)->GetPrimitiveArrayCritical(env, postUsedArray, &isCopy);
+		if (NULL == postUsedArrayElems) {
+			goto fail2;
+		}
+		postCommittedArrayElems = (jlong*) (*env)->GetPrimitiveArrayCritical(env, postCommittedArray, &isCopy);
+		if (NULL == postCommittedArrayElems) {
+			goto fail2;
+		}
+		postMaxArrayElems = (jlong*) (*env)->GetPrimitiveArrayCritical(env, postMaxArray, &isCopy);
+		if (NULL == postMaxArrayElems) {
+			goto fail2;
+		}
+		omrthread_rwmutex_enter_read(mgmt->managementDataLock);
+		for (idx = 0; idx < gcInfo->arraySize; idx++) {
+			initialArrayElems[idx]       = gcInfo->initialSize[idx];
+			preUsedArrayElems[idx]       = gcInfo->preUsed[idx];
+			preCommittedArrayElems[idx]  = gcInfo->preCommitted[idx];
+			preMaxArrayElems[idx]        = gcInfo->preMax[idx];
+			postUsedArrayElems[idx]      = gcInfo->postUsed[idx];
+			postCommittedArrayElems[idx] = gcInfo->postCommitted[idx];
+			postMaxArrayElems[idx]       = gcInfo->postMax[idx];
```

Comment:
Please don't try to line up the = signs like this.  The first time this code is refactored, all the lines will need to be modified to keep the alignment which makes blame harder

Reply:
done